src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -469,7 +469,11 @@ private IQueryCache BuildQueryCache(string queryCacheName)
 
 		public EventListeners EventListeners
 		{
-			get { return eventListeners; }
+			get
+			{
+				CheckNotDisposed();
+				return eventListeners;
```

Comment:
The session factory closing `Destroy` the event listeners, so better guard against any later use of them.